### PR TITLE
FI-1082 disabled interactive sudo and su connections

### PR DESCRIPTION
**Pull request by Karthigaiselvan**

### CODE REVIEW

File: src/main/java/com/xebialabs/overthere/ssh/ReplacingInputStream.java [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    }
+
+    /**
+     * Replace occurrences of pattern in the input. Note: input is assumed to be UTF-8 encoded. If not the case use byte[] based pattern and replacement.
+     * @param in input
+     * @param pattern pattern to replace.
+     * @param replacement the replacement or null
+     */
+    public ReplacingInputStream(InputStream in, String pattern, String replacement) {
+        this(in, pattern.getBytes(UTF_8), replacement==null ? null : replacement.getBytes(UTF_8));
```
Comment (racheruk) [inline src/main/java/com/xebialabs/overthere/ssh/ReplacingInputStream.java:44]: Suggest adding a null check for _pattern_ too

**Comment by racheruk** [comment]
> I'd really like to see failing tests for the two use cases which concern us: the multi match problem and the remote process leading stdin to stdout. Should we stub/mock SshProcess and test SshElevatedUserConnection#createProcess so as to exercise our new stream handing? It would be more of an integration test than a unit test but would allow us to show that we can reproduce and fix the behavior.

**Comment by racheruk** [comment]
> I'd really like to see failing tests for the two use cases which concern us: the multi match problem and the remote process leading stdin to stdout. Should we stub/mock SshProcess and test SshElevatedUserConnection#createProcess so as to exercise our new stream handing? It would be more of an integration test than a unit test but would allow us to show that we can reproduce and fix the behavior.

@jpshackelford In general, stdin (i.e. text/passwords that overthere sends) always endsup in stdout (i.e. the log file of the total interaction). Thats how ssh interaction with overthere works.

While adding an integration test is good, i would vote to write bunch of unit tests that will make sure that the state inside the new stream instance is as expected while replacing the password.

**Review by racheruk — APPROVED** [review]

**Comment by Karthigaiselvan** [comment]
Unit test case covered for ReplacingInputStream instance with replaceText, multipleOccurrenceReplace, notMatch, caseSentive and exception scenarios. inputstream bytes compared with pattern bytes in stream state. Stream state will be change on reading and comparing each bytes. Always the final state will be NOT_MATCHED. 
 
 ex: 
 InputStream = "echo Password:"
 Pattern = "Password"
Replacement = "Test" 

here in ReplacingInputStream the initial state will be NOT_MATCHED and the bytes will be compared with pattern bytes. 
If matches found the state will be "MATCHING", if the pattern fully matched then the state changed to "REPLACING" state and replacement done. Again the state will be changed to "NOT_MATCHED" state. So the final state will be "NOT_MATCHED" after the replacement.

**Pull request merged**